### DISCUSSION TITLE
Why is it "flipdna" not "revcompdna" or so?

### DISCUSSION THREAD

**Discussion by jan-glx** [yachielab/QUEEN#4] [General] [answered]
Why is it "flipdna" not "revcompdna" or so?
Reverse complement seem to be the standard expression used in click-and-point and cli-based in-silico-molecular-cloning-software (e.g. pydna, Geneiuous, ugene)

**Comment by ponnhide** [comment, marked as answer]
Thank you for your comment.
In QUEEN, the operational function names ("cutdna", "modifyends", "flipdna", "joindna") have been determined to represent actual DNA sequence manipulations. For example, I believe the "flipdna" name is more appropriate than "revcompdna" when describing DNA inversion by site-specific recombination, such as Cre/loxP recombination.

I know that there are cases in which users may want to get a reverse complementary sequence regardless of experimental DNA sequence manipulations. However, I did not use a name like "revcompdna" because it is unsuitable for the QUEEN concept.

**Reply by jan-glx** [reply]
I really like how QUEEN tries to abstracts all the stuff we can do in the lab into a few operations and how it enables reproducible, modifiable and appealing descriptions of those.

The following might appear pedantic, but I'd really like to fully understand what a QUEEN objects and operational functions are meant to represent.

Flipping or reverse complementing an actual _dsDNA molecule_ would be a null operation (It's the same molecule no matter which strand you define as top strand).

QUEEN operational functions seem to operate on _dsDNA sequences_ i.e. `joindna` requires the sequences to be oriented correctly (it doesn't find the matching end like a ligase would).

I'd argue that `flipdna` doesn't represent something like DNA inversion by site-specific recombination (as it would then need to operate only on a subset the sequence) but that it rather really only changes which strand is considered the top strand (for visualization and for defining which site the other operations act on).  Thus I see "reverse complement" as a more appropriate name.

Do you have plans to provide higher level functions, representing reactions like PCR, in addition to the low level interface?

**Reply by ponnhide** [reply]
> QUEEN operational functions seem to operate on dsDNA sequences i.e. joindna requires the sequences to be oriented correctly (it doesn't find the matching end like a ligase would).

I could understand your argument from this statement. As you pointed out, the current `joindna` function requires orient correctness and does not match the DNA sequence manipulation in molecular cloning. OK, I will consider operational function names and implementations more to represent their actual functions.

> Do you have plans to provide higher level functions, representing reactions like PCR, in addition to the low level interface?

At present, I don't have such a plan because QUEEN is designed to represent various molecular cloning processes by combining operational functions. However, recently, I've thought that such high-level interfaces may be required to promote the use of QUEEN in the life science community.

**Reply by jan-glx** [reply]
Happy to see that you added such high-level functions in 1.2 🤩
